src/librustc_yk_sections/mir_cfg.rs

```diff
@@ -96,9 +95,7 @@ impl<'a, 'tcx, 'gcx> ConvCx<'a, 'tcx, 'gcx> {
     /// Add `bb` as a definition site of the TIR variable `var`.
     fn push_def_site(&self, bb: BasicBlock, var: TirVarIndex) {
         let mut sites = self.def_sites.borrow_mut();
-        // Vector indices are `usize`, but variable indices are `u32`, so converting from a
-        // variable index to a vector index is always safe if a `usize` can express all `u32`s.
-        debug_assert!(size_of::<usize>() >= size_of::<u32>());
+        // `var` is `u32`. The conversion up to `usize` was checked above in `tir_var()`.
```

Comment:
But is `tir_var` guaranteed to have been called before `push_def_site`? I think so, at which point something like "This conversion is safe because TirVarIndex was generated by tir_var which guarantees that a u32 can fit in a usize" is probably the right summary.

Reply:
I've just double checked. We convert the "Place" first, which makes a TIR variable, and then after that we push a def site.

I'll update the comment.

Reply:
OK.